From 38102f2b39d49e52c7a7c51c0fb340b3e70362d2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Peter=20Sz=C3=BCcs?= <peter.szucs.dev@gmail.com>
Date: Sun, 3 Aug 2014 00:28:59 +0200
Subject: [PATCH] Add support for Android Standalone toolchains.

---
 Alc/backends/android.c | 4 ++--
 CMakeLists.txt         | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/Alc/backends/android.c b/Alc/backends/android.c
index 2201612..842f60b 100644
--- a/Alc/backends/android.c
+++ b/Alc/backends/android.c
@@ -12,7 +12,7 @@ JavaVM *alcGetJavaVM(void) {
 	return javaVM;
 }
 
-jint JNI_OnLoad(JavaVM *vm, void *reserved) {
+jint JNI_OnLoad_OpenAL(JavaVM *vm, void *reserved) {
 	// BackendFuncs func_list;
 	if (androidOpenALFuncs.alc_android_set_java_vm) {
 		androidOpenALFuncs.alc_android_set_java_vm(vm);
@@ -21,7 +21,7 @@ jint JNI_OnLoad(JavaVM *vm, void *reserved) {
 	return JNI_VERSION_1_4;
 }
 
-void JNI_OnUnload (JavaVM *vm, void *reserved)
+void JNI_OnUnload_OpenAL(JavaVM *vm, void *reserved)
 {
 	if (androidOpenALFuncs.alc_android_set_java_vm) {
 		androidOpenALFuncs.alc_android_set_java_vm(NULL);
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 81c0980..8bb19cc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -737,9 +737,9 @@ IF(ALSOFT_BACKEND_OPENSL)
         CHECK_SHARED_FUNCTION_EXISTS(slCreateEngine "SLES/OpenSLES.h" OpenSLES "" HAVE_LIBOPENSLES)
         IF(HAVE_LIBOPENSLES)
             SET(HAVE_OPENSL 1)
-            SET(ALC_OBJS  ${ALC_OBJS} Alc/backends/opensl.c)
+            SET(ALC_OBJS  ${ALC_OBJS} Alc/backends/opensl.c Alc/backends/audiotrack.c Alc/backends/android.c)
             SET(BACKENDS  "${BACKENDS} OpenSL,")
-            SET(EXTRA_LIBS OpenSLES ${EXTRA_LIBS})
+            SET(EXTRA_LIBS OpenSLES log ${EXTRA_LIBS})
         ENDIF()
     ENDIF()
 ENDIF()
-- 
1.9.1

