diff --git a/OgreMain/src/OgreInstanceBatch.cpp b/OgreMain/src/OgreInstanceBatch.cpp
index bbcc55acb..0dc74ae79 100644
--- a/OgreMain/src/OgreInstanceBatch.cpp
+++ b/OgreMain/src/OgreInstanceBatch.cpp
@@ -162,6 +162,9 @@ namespace Ogre
 
 
         mBoundingRadius = Math::boundingRadiusFromAABBCentered( mFullBoundingBox );
+        if (mParentNode) {
+            mParentNode->needUpdate();
+        }
         mBoundsDirty    = false;
     }
 
