*** a/RenderSystems/GL/src/OSX/OgreOSXCocoaWindow.mm	Thu Aug  4 14:35:02 2011
--- b/RenderSystems/GL/src/OSX/OgreOSXCocoaWindow.mm	Thu Aug  4 14:17:46 2011
***************
*** 205,210 ****
--- 205,226 ----
  			NSOpenGLContext *shareContext = mainContext == 0? nil : mainContext->getContext();
              mGLContext = [[NSOpenGLContext alloc] initWithFormat:mGLPixelFormat shareContext:shareContext];
  
+ 
+ bool useCurrentGLContext = false;
+ if(miscParams) {
+ 	NameValuePairList::const_iterator opt = miscParams->find("currentGLContext");
+ 	if (opt != miscParams->end()){
+ 		useCurrentGLContext = StringConverter::parseBool(opt->second);
+ 	}
+ }
+          
+ if(useCurrentGLContext) {
+ 	LogManager::getSingleton().logMessage("Mac Cocoa Window: Rendering on the current OpenGL context");
+ 	mView = [[NSOpenGLContext currentContext] view];
+ 	[mGLContext setView:mView];
+ 	WindowEventUtilities::_addRenderWindow(this);
+ } else {
+ 
              if(miscParams)
                  opt = miscParams->find("externalWindowHandle");
              
***************
*** 243,248 ****
--- 259,265 ----
                  // Add our window to the window event listener class
                  WindowEventUtilities::_addRenderWindow(this);
              }
+ }
  
              [mGLContext makeCurrentContext];
  
